SharkyCTF2020/Blockchain/Guessing.md

```diff
@@ -0,0 +1,69 @@
+# Warmup
+
+## Setup
+
+The challenge is available at http://ethereum.sharkyctf.xyz/level/2
+
+Before starting, it is recommended to follow the suggested steps at http://ethereum.sharkyctf.xyz/help to setup Metamask, get Ether from a Ropsten faucet, and Remix IDE.
+
+## Challenge
+
+The contract to attack is the following :
+
+```solidity
+pragma solidity = 0.4.25;
+
+contract Guessing {
+    address public owner;    
+    bytes32 private passphrase;
+    
+    constructor(bytes32 _passphrase) public payable {
+        owner = msg.sender;
+        passphrase = keccak256(abi.encodePacked(_passphrase));
+    }
+    
+    function withdraw() public {
+        require(msg.sender == owner);
+        msg.sender.call.value(address(this).balance)();
+    }
+
+    function claim(bytes32 _secret) public payable {
+        require(keccak256(abi.encodePacked(_secret)) == passphrase);
+        owner = msg.sender; 
+    }
+}
+```
+
+The goal is then to be able to call `withdraw()` from the contract.
+
+## Hint
+
+There seems to be a passphrase needed to unlock the contract, and it is hashed by keccak256, so it seems to be secure. Nonetheless, are transactions on a public blockchain readable ?
+
+## Solution
+
+All transactions on a public blockchain are recorded and can read by anyone, including the first transaction that created a contract. Therefore, if a secret is passed to a contract at its creation, one should be able to read what it was just by doing at the records of the initial transaction.
```

Comment:
can read -> can be read
Pas très clair la phrase "just by doing at the records of the initial transaction."
C'était peut être "just by looking" ?